core/java/src/org/openda/exchange/AbstractDataObject.java

```diff
@@ -0,0 +1,29 @@
+package org.openda.exchange;
+
+import org.openda.interfaces.IDataObject;
+import org.openda.interfaces.IExchangeItem;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public abstract class AbstractDataObject implements IDataObject {
+
+	public static final String[] EMPTY_STRING_ARRAY = new String[0];
+	protected Map<String, IExchangeItem> exchangeItems = new LinkedHashMap<>();
+
+	@Override
+	public String[] getExchangeItemIDs() {
+		return this.exchangeItems.keySet().toArray(EMPTY_STRING_ARRAY);
+	}
+
+	@Override
```

Comment:
This methods should check the role of the exchange item. However is only used in model_wflow.